pkg/tests/detailed_diff_unknown_test.go

```diff
@@ -900,7 +900,7 @@ func TestUnknownCollectionForceNewDetailedDiff(t *testing.T) {
 			runTest(t, program2, autogold.Expect(`
     + prov:index/aux:Aux: (create)
         [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
-    +-prov:index/test:Test: (replace)
+    ~ prov:index/test:Test: (update)
```

Comment:
It seems that TF is optimistic about replacing ForceNew properties with unknown and we were being pessimistic. This change ensures we match TF and ForceNew proeprties replaced with unknowns are now predicted NOT to cause a replacement.

We can add a workaround if we disagree here.